src/api/spec/controllers/webui/comments_controller_spec.rb

```diff
@@ -106,4 +106,95 @@
       it { expect(Comment.where(id: other_comment.id)).to eq([]) }
     end
   end
+
+  describe 'EDIT update' do
+    let(:project) { create(:project) }
+    let(:package) { create(:package, project: project) }
+    let(:bs_request) { create(:set_bugowner_request) }
+    let(:admin_user) { create(:admin_user, login: 'Admin') }
+    let(:comment) { create(:comment_project, user: user) }
+    let(:other_comment) { create(:comment_project) }
+
+    context 'with invalid commentable_type' do
+      let(:comment_params) do
+        { comment: { body: 'This is AWESOME!' }, commentable_type: 'FOOBAR', commentable_id: 31_337 }
+      end
+
+      subject { post :create, params: comment_params }
+
+      it { expect(subject.request.flash[:success]).to be_nil }
+      it { expect(subject.request.flash[:error]).not_to(be_nil) }
+      it { expect(subject).to redirect_to(root_path) }
+    end
+
+    context 'with a valid comment' do
+      RSpec.shared_examples 'updating a comment' do
+        before do
+          params = { id: comment.id, comment: { body: "This #{commentable.model_name.singular} is AWFUL!" } }
+          put :update, params: params
+        end
+
+        it { expect(flash[:success]).to eq('Comment updated successfully.') }
+        it { expect(comment.reload.body).to eq("This #{commentable.model_name.singular} is AWFUL!") }
+      end
+
+      context 'only XML requests' do
+        let(:commentable) { project }
+
+        let(:params) do
+          { id: comment.id, comment: { body: "This #{commentable.model_name.singular} is AWFUL!" } }
+        end
+
+        it 'responds to html requests' do
+          put :update, params: params, format: :html
+          expect(response.header['Content-Type']).to include 'text/html'
+        end
+
+        it 'does not respond to other json requests' do
+          expect { put :update, params: params, format: :json }.to raise_error(ActionController::UnknownFormat)
+        end
+
+        it 'does not respond to json requests' do
```

Comment:
Those are `xml requests` instead, right?

Reply:
🤦 yup, on it